### PR TITLE
[two_dimensional_scrollables] Fix paint bug when rows/columns are pinned and axes are reversed

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/135386

This adds one more golden to verify the painting, but once public mock_canvas rolls to stable in flutter_test, we can remove all of the goldens and just verify using the `paints` method.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/two_dimensional_scrollables/test/table_view/goldens/tableSpanDecoration.defaultMainAxis.png
Comment (goderbauer) [inline packages/two_dimensional_scrollables/test/table_view/goldens/tableSpanDecoration.defaultMainAxis.png:1]: Just wondering why this one and the next one is included in this PR as "changed". Aren't they 100% identical?

File: packages/two_dimensional_scrollables/test/table_view/table_test.dart [hunk 1]
```diff
@@ -1040,6 +1040,59 @@ void main() {
...
+        columnCount: 2,
+        pinnedColumnCount: 1,
+        columnBuilder: (int index) => TableSpan(
+          extent: const FixedTableSpanExtent(200.0),
+          foregroundDecoration: const TableSpanDecoration(
+              border: TableSpanBorder(
+                  trailing: BorderSide(
+            color: Colors.orange,
+            width: 3,
+          ))),
```
Comment (goderbauer) [inline packages/two_dimensional_scrollables/test/table_view/table_test.dart:1062]: formatting nit (probably just missing some trailing commas?):

```suggestion
          foregroundDecoration: const TableSpanDecoration(
            border: TableSpanBorder(
              trailing: BorderSide(
                color: Colors.orange,
                width: 3,
              ),
            ),
          ),
```

File: packages/two_dimensional_scrollables/test/table_view/table_test.dart [hunk 2]
```diff
@@ -1040,6 +1040,59 @@ void main() {
...
+          ),
+        ),
+        rowBuilder: (int index) => TableSpan(
+          extent: const FixedTableSpanExtent(200.0),
+          foregroundDecoration: const TableSpanDecoration(
+              border: TableSpanBorder(
+                  leading: BorderSide(
+            color: Colors.green,
+            width: 3,
+          ))),
```
Comment (goderbauer) [inline packages/two_dimensional_scrollables/test/table_view/table_test.dart:1074]: same nit here

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/two_dimensional_scrollables/test/table_view/goldens/tableSpanDecoration.defaultMainAxis.png
Comment (Piinks) [reply packages/two_dimensional_scrollables/test/table_view/goldens/tableSpanDecoration.defaultMainAxis.png:1]: Yeah I was curious about this too. Probably really minor pixel diffs. Running `flutter test --update-goldens` output new images per git when I ran it for the new golden test here, so I figured I would let them be updated.

File: packages/two_dimensional_scrollables/test/table_view/table_test.dart (hunk 1 above)
Comment (Piinks) [reply packages/two_dimensional_scrollables/test/table_view/table_test.dart]: Good catch, I found a couple more cases of this, fixed them all. :)

**Pull request merged**